jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -1867,8 +1867,14 @@ protected void recycle()
         _async = null;
         _asyncNotSupportedSource = null;
         _handled = false;
+        _attributes = Attributes.unwrap(_attributes);
         if (_attributes != null)
-            _attributes.clearAttributes();
+        {
+            if (AttributesMap.class.equals(_attributes.getClass()))
```

Comment:
Because classes that extend AttributesMap with unknown behaviours will pass an instanceof test.
I  already know it is an instanceof `AttributesMap` because that is the type of the field.  I want to ensure it is a base implementation that we can clear and not have any special behaviour left over.